File: Week 01/id_503/leetcode_1_503.go
package week01

// 引入 map, 空间换时间
func twoSum(nums []int, target int) []int {

	subToIndex := map[int]int{}

	for i, e := range nums {
		sub := target - e
		if index, ok := subToIndex[sub]; ok {
			return []int{index, i}
		}

		subToIndex[e] = i
	}

	return nil
}

File: Week 01/id_503/leetcode_1_503.js
/**
 * 引入 map, 空间换时间
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
	const map = new Map();

	for (let i = 0; i < nums.length; i++) {
		const key = target - nums[i];
		const value = map.get(key);
		if (value !== undefined) {
			return [value, i];
		}

		map.set(nums[i], i);
	}

	return []
};